File: backend/app/api/draw.py
"""M2 AI绘图服务 - API 路由"""

from __future__ import annotations
import uuid

from fastapi import APIRouter, HTTPException, Query

from app.models.draw import (
    RenderTaskRequest, RenderTask, RenderTaskList,
    BatchRenderRequest, PipelineConfig, ModelInfo, WorkflowTemplate,
)
from app.services.draw.renderer import renderer

router = APIRouter()


@router.post("/render", response_model=RenderTask)
async def create_render_task(req: RenderTaskRequest):
    """F2.1 文生图 / F2.2 图生图 — 创建渲染任务"""
    task = renderer.create_task(req)
    return task


@router.post("/batch-render")
async def batch_render(req: BatchRenderRequest):
    """F2.3 批量生成 — 提交多个渲染任务"""
    task_ids = []
    for task_req in req.tasks:
        task = renderer.create_task(task_req)
        task_ids.append(task.task_id)
    return {"task_ids": task_ids, "count": len(task_ids)}


@router.get("/tasks", response_model=RenderTaskList)
async def list_tasks(
    project_id: str | None = Query(None),
    page: int = Query(1, ge=1),
    size: int = Query(20, ge=1, le=100),
):
    """获取渲染任务列表"""
    items = renderer.list_tasks(project_id, page, size)
    total = renderer.get_task_count(project_id)
    return RenderTaskList(items=items, total=total, page=page, size=size)


@router.get("/tasks/{task_id}", response_model=RenderTask)
async def get_task(task_id: str):
    """查询单个任务状态"""
    task = renderer.get_task(task_id)
    if not task:
        raise HTTPException(status_code=404, detail="任务不存在")
    return task


@router.get("/pipelines", response_model=list[PipelineConfig])
async def list_pipelines():
    """获取支持的渲染管线"""
    return renderer.get_pipelines()


@router.get("/pipelines/{pipeline_id}", response_model=PipelineConfig)
async def get_pipeline(pipeline_id: str):
    pipe = renderer.get_pipeline(pipeline_id)
    if not pipe:
        raise HTTPException(status_code=404, detail="管线不存在")
    return pipe


@router.get("/models", response_model=list[ModelInfo])
async def list_models(model_type: str | None = Query(None)):
    """获取模型列表"""
    return renderer.get_models(model_type)


@router.get("/workflows", response_model=list[WorkflowTemplate])
async def list_workflows(workflow_type: str | None = Query(None)):
    """获取工作流模板"""
    return renderer.get_workflows(workflow_type)

File: frontend/src/api/draw.ts
/** M2 AI绘图服务 - API服务层 */

import type { RenderTaskRequest, RenderTask, RenderTaskList, PipelineConfig, ModelInfo, WorkflowTemplate } from './drawTypes'

const BASE_URL = 'http://localhost:8000/api/v1/draw'

async function request<T>(url: string, options?: RequestInit): Promise<T> {
  const res = await fetch(url, {
    headers: { 'Content-Type': 'application/json' },
    ...options,
  })
  if (!res.ok) {
    const err = await res.json().catch(() => ({ detail: res.statusText }))
    throw new Error(err.detail || `HTTP ${res.status}`)
  }
  return res.json()
}

/** 创建渲染任务 */
export async function createRenderTask(req: RenderTaskRequest): Promise<RenderTask> {
  return request<RenderTask>(`${BASE_URL}/render`, {
    method: 'POST',
    body: JSON.stringify(req),
  })
}

/** 批量渲染 */
export async function batchRender(tasks: RenderTaskRequest[]): Promise<{ task_ids: string[]; count: number }> {
  return request(`${BASE_URL}/batch-render`, {
    method: 'POST',
    body: JSON.stringify({ project_id: tasks[0]?.project_id || 'default', tasks }),
  })
}

/** 获取任务列表 */
export async function getTasks(projectId?: string, page = 1, size = 20): Promise<RenderTaskList> {
  const params = new URLSearchParams()
  if (projectId) params.set('project_id', projectId)
  params.set('page', String(page))
  params.set('size', String(size))
  return request(`${BASE_URL}/tasks?${params}`)
}

/** 查询任务状态 */
export async function getTask(taskId: string): Promise<RenderTask> {
  return request(`${BASE_URL}/tasks/${taskId}`)
}

/** 获取管线列表 */
export async function getPipelines(): Promise<PipelineConfig[]> {
  return request(`${BASE_URL}/pipelines`)
}

/** 获取模型列表 */
export async function getModels(modelType?: string): Promise<ModelInfo[]> {
  const params = modelType ? `?model_type=${modelType}` : ''
  return request(`${BASE_URL}/models${params}`)
}

/** 获取工作流模板 */
export async function getWorkflows(workflowType?: string): Promise<WorkflowTemplate[]> {
  const params = workflowType ? `?workflow_type=${workflowType}` : ''
  return request(`${BASE_URL}/workflows${params}`)
}
